tower-filter/src/lib.rs

```diff
@@ -21,17 +21,17 @@ pub struct Filter<T, U> {
     counts: Arc<Counts>,
 }
 
-pub struct ResponseFuture<T, S>
-where S: Service,
+pub struct ResponseFuture<T, S, R>
+where S: Service<R>,
 {
-    inner: Option<ResponseInner<T, S>>,
+    inner: Option<ResponseInner<T, S, R>>,
 }
 
 #[derive(Debug)]
-struct ResponseInner<T, S>
-where S: Service,
+struct ResponseInner<T, S, R>
+where S: Service<R>,
```

Comment:
Again, we need to maintain the `S: Service` bound, since the struct definition uses `S::Future`, so the additional request type parameter was added.